twoliter/src/project/lock/mod.rs

```diff
@@ -200,6 +204,43 @@ impl Lock {
         }
     }
 
+    fn validate_version_source_consistency(&self) -> Result<()> {
+        Self::validate_locked_image_consistency(&self.sdk, "SDK")?;
+
+        for (index, kit) in self.kit.iter().enumerate() {
+            Self::validate_locked_image_consistency(
+                kit,
+                &format!("kit[{}] ({})", index, kit.name),
+            )?;
+        }
+
+        Ok(())
+    }
+
+    fn validate_locked_image_consistency(image: &LockedImage, context: &str) -> Result<()> {
```

Comment:
We should find a different way to represent `context` that isn't strings.

Is there an existing enum type we can use to disambiguate SDK images from Kit images? Or perhaps we should just remove the `context` argument and rely on `LockedImage` having a reasonable implementation of `Display` (which it does!)